app/controllers/miq_policy_controller/policies.rb

```diff
@@ -1,91 +1,113 @@
 module MiqPolicyController::Policies
   extend ActiveSupport::Concern
 
-  def policy_edit
-    case params[:button]
-    when "cancel"
-      id = params[:id] ? params[:id] : "new"
-      return unless load_edit("policy_edit__#{id}", "replace_cell__explorer")
-      @policy = MiqPolicy.find_by(:id => @edit[:policy_id]) if @edit[:policy_id]
-      if @policy && @policy.id
-        add_flash(_("Edit of %{model} \"%{name}\" was cancelled by the user") % {:model => ui_lookup(:model => "MiqPolicy"), :name => @policy.description})
-      else
-        add_flash(_("Add of new %{models} was cancelled by the user") % {:models => ui_lookup(:model => "MiqPolicy")})
+  def policy_edit_cancel
+    id = params[:id] ? params[:id] : "new"
+    return unless load_edit("policy_edit__#{id}", "replace_cell__explorer")
+    @policy = MiqPolicy.find_by(:id => @edit[:policy_id]) if @edit[:policy_id]
+    if @policy && @policy.id
+      add_flash(_("Edit of %{model} \"%{name}\" was cancelled by the user") % {:model => ui_lookup(:model => "MiqPolicy"), :name => @policy.description})
+    else
+      add_flash(_("Add of new %{models} was cancelled by the user") % {:models => ui_lookup(:model => "MiqPolicy")})
+    end
+    @edit = nil
+    get_node_info(x_node)
+    replace_right_cell(:nodetype => @nodetype)
+  end
+
+  def policy_edit_reset
+    @sb[:action] = "policy_edit"
+    policy_build_edit_screen(session[:edit].try(:key?, :typ) ? session[:edit][:typ] : params[:typ])
+    if params[:button] == "reset"
+      add_flash(_("All changes have been reset"), :warning)
```

Comment:
nitpick: please use oneliner if messages in the same line.

Reply:
yes, that could have been done if more refactoring was done